sys/kern/vm_object.c

```diff
@@ -55,43 +55,39 @@ void vm_object_add_page(vm_object_t *obj, off_t offset, vm_page_t *pg) {
   }
 }
 
-static void vm_object_remove_page_nolock(vm_object_t *obj, vm_page_t *page) {
-  page->offset = 0;
-  page->object = NULL;
+static void vm_object_remove_pages_nolock(vm_object_t *obj, vm_offset_t offset,
+                                          size_t length) {
+  assert(mtx_owned(&obj->mtx));
+  assert(page_aligned_p(offset) && page_aligned_p(length));
 
-  TAILQ_REMOVE(&obj->list, page, obj.list);
-  vm_page_free(page);
-  obj->npages--;
+  vm_page_t *pg, *next;
+  TAILQ_FOREACH_SAFE (pg, &obj->list, obj.list, next) {
+    if (pg->offset >= offset + length)
+      break;
+    if (pg->offset < offset)
+      continue;
+
+    pg->offset = 0;
+    pg->object = NULL;
+    TAILQ_REMOVE(&obj->list, pg, obj.list);
+    vm_page_free(pg);
+    obj->npages--;
+  }
 }
 
-void vm_object_remove_page(vm_object_t *obj, vm_page_t *page) {
+void vm_object_remove_pages(vm_object_t *obj, vm_offset_t off, size_t len) {
   SCOPED_MTX_LOCK(&obj->mtx);
-
-  vm_object_remove_page_nolock(obj, page);
+  vm_object_remove_pages_nolock(obj, off, len);
 }
 
-void vm_object_remove_range(vm_object_t *object, off_t offset, size_t length) {
-  SCOPED_MTX_LOCK(&object->mtx);
-
-  vm_page_t *pg, *next;
-  TAILQ_FOREACH_SAFE (pg, &object->list, obj.list, next) {
-    if (pg->offset >= (off_t)(offset + length))
-      break;
-    if (pg->offset >= offset)
-      vm_object_remove_page_nolock(object, pg);
-  }
-}
+#define vm_object_remove_all_pages(obj)                                        \
+  vm_object_remove_pages((obj), 0, -PAGESIZE)
```

Comment:
I find this macro confusing. I presume that `-PAGESIZE` will be positive, but it doesn't look so obvious. And is there a reason to use macros instead of `static inline`?

Reply:
I can cast `-PAGESIZE` to `size_t` if that helps. This is a way to express succinctly the largest number aligned to `PAGESIZE` boundary.

Reply:
Added a cast to `size_t`.